### PR TITLE
Add cloudinaryUploader HOC

### LINKED ISSUE #1: Add cloudinaryUploader HOC
When it's needed on another project.

### PR DESCRIPTION
**Pull request by rkillackey**
Resolves #1

Restructured `utils` a bit, lmk what you think.

### CODE REVIEW

**Comment by rkillackey** [comment]
@dpikt One sec-- let me add a test for failed upload!

**Comment by rkillackey** [comment]
@dpikt Ready for review!

File: src/utils/remove-extension.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// Removes file extension from file name if asset is an image
```
Comment (dpikt) [inline src/utils/remove-extension.js]: This comment doesn't quite apply anymore (I think I did this!)

File: test/cloudinary-uploader.test.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    expect(responseJson.upload_preset).toEqual('default')
+  })
+})
+
+test('cloudinaryUploader throws an error if request fails', () => {
+  const Wrapped = () => <h1>Hi</h1>
+  const Wrapper = cloudinaryUploader({ ...props, endpoint: '/failure' })(Wrapped)
+  const component = shallow(<Wrapper />)
+  const { upload } = component.props()
+  expect(() => upload(fileData, file).toThrow())
```
Comment (dpikt) [inline test/cloudinary-uploader.test.js]: I don't think you can use `toThrow()` with async errors. You should call `upload().catch()` with an assertion in the catch statement. 

You'll also have to use [expect.assertions(1)](https://facebook.github.io/jest/docs/en/expect.html#expectassertionsnumber) to make sure the assertion is called.

File: test/cloudinary-uploader.test.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+})
+
+test('cloudinaryUploader updates the `uploadStatus` prop if request fails', () => {
+  const Wrapped = () => <h1>Hi</h1>
+  const Wrapper = cloudinaryUploader({ ...props, endpoint: '/failure' })(Wrapped)
+  const component = shallow(<Wrapper />)
+  const { upload } = component.props()
+  return upload(fileData, file).catch(() => {
```
Comment (dpikt) [inline test/cloudinary-uploader.test.js]: Use `expect.assertions` here too

File: src/utils/remove-extension.js (hunk 1 above)
Comment (rkillackey) [reply src/utils/remove-extension.js]: Oops, that was me, just forgot to remove it

**Comment by rkillackey** [comment]
@dpikt Ready for another look!

**Pull request merged**